File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/styles/sources/RCTMGLRasterSourceManager.java
package com.mapbox.rctmgl.components.styles.sources;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

import java.util.Map;

import javax.annotation.Nonnull;

public class RCTMGLRasterSourceManager extends RCTMGLTileSourceManager<RCTMGLRasterSource> {
    public static final String REACT_CLASS = "RCTMGLRasterSource";

    public RCTMGLRasterSourceManager(ReactApplicationContext reactApplicationContext) {
        super(reactApplicationContext);
    }

    @Nonnull
    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Nonnull
    @Override
    protected RCTMGLRasterSource createViewInstance(@Nonnull ThemedReactContext reactContext) {
        return new RCTMGLRasterSource(reactContext);
    }

    @ReactProp(name="tileSize")
    public void setTileSize(RCTMGLRasterSource source, int tileSize) {
        source.setTileSize(tileSize);
    }

    @Override
    public Map<String, String> customEvents() {
        return null;
    }
}

File: android/src/main/java-v10/com/mapbox/rctmgl/components/styles/sources/RCTMGLRasterSourceManager.kt
package com.mapbox.rctmgl.components.styles.sources

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp
import javax.annotation.Nonnull

class RCTMGLRasterSourceManager(reactApplicationContext: ReactApplicationContext) :
    RCTMGLTileSourceManager<RCTMGLRasterSource?>(reactApplicationContext) {
    @Nonnull
    override fun getName(): String {
        return REACT_CLASS
    }

    @Nonnull
    override fun createViewInstance(@Nonnull reactContext: ThemedReactContext): RCTMGLRasterSource {
        return RCTMGLRasterSource(reactContext)
    }

    @ReactProp(name = "tileSize")
    fun setTileSize(source: RCTMGLRasterSource, tileSize: Int) {
        source.setTileSize(tileSize)
    }

    override fun customEvents(): Map<String, String>? {
        return null
    }

    companion object {
        const val REACT_CLASS = "RCTMGLRasterSource"
    }
}